logprep/util/rule_dry_runner.py

```diff
@@ -44,146 +44,86 @@
-    ----------
-    patched_runner : Runner
-        The patched logprep runner
-
-    Returns
-    -------
-    parsed_outputs : list
-        A list of logprep outputs containing events, extra outputs like pre-detections or pseudonyms
-        and errors
-    """
-    parsed_outputs = [None, None, None]
-    output_config = list(patched_runner._configuration.get("output").values())[0]
-    output_paths = [
-        output_path for key, output_path in output_config.items() if "output_file" in key
-    ]
-
-    for output_path in output_paths:
-        remove_file_if_exists(output_path)
-
-    patched_runner.start()
-
-    for index, output_path in enumerate(output_paths):
-        parsed_outputs[index] = parse_jsonl(output_path)
-        remove_file_if_exists(output_path)
-
-    return parsed_outputs
-
-
-def get_patched_runner(config_path, logger):
-    """
-    Creates a patched runner that bypasses check to obtain non singleton instance and the runner
-    won't continue iterating on an empty pipeline.
-
-    Parameters
-    ----------
-    config_path : str
-        The logprep configuration that should be used for the patched runner
-    logger : Logger
-        The application logger the runner should use
-
-    Returns
-    -------
-    runner : Runner
-        The patched logprep runner
-    """
-    runner = Runner(bypass_check_to_obtain_non_singleton_instance=True)
-    runner.set_logger(logger)
-    runner.load_configuration(config_path)
-
-    # patch runner to stop on empty pipeline
-    def keep_iterating():
-        """generator that stops on first iteration"""
-        return  # nosemgrep
-        yield
-
-    runner._keep_iterating = keep_iterating  # pylint: disable=protected-access
-
-    return runner
-
-
 class DryRunner:
     """Used to run pipeline with given events and show changes made by processing."""
 
-    def __init__(self, dry_run: str, config_path: str, full_output: bool, use_json: bool, logger):
-        self._tmp_path = tempfile.mkdtemp()
-        self.patched_config_path = Configuration.patch_yaml_with_json_connectors(
-            original_config_path=config_path, output_dir=self._tmp_path, input_file_path=dry_run
+    @cached_property
+    def _tmp_path(self):
+        return tempfile.mkdtemp()
+
+    @cached_property
+    def _pipeline(self):
+        patched_config_path = Configuration.patch_yaml_with_json_connectors(
+            original_config_path=self._config_path,
+            output_dir=self._tmp_path,
+            input_file_path=self._input_file_path,
+        )
+        config = Configuration.create_from_yaml(patched_config_path)
+        config.verify_pipeline_without_processor_outputs(self._logger)
+        del config["output"]
+        return Pipeline(config=config)
+
+    @cached_property
+    def _input_documents(self):
+        return (
+            parse_json(self._input_file_path)
+            if self._use_json
+            else parse_jsonl(self._input_file_path)
         )
```

Comment:
```suggestion
        return (
            GetterFactory.from_string(self._input_file_path).get_yaml()
            if self._use_json
            else GetterFactory.from_string(self._input_file_path).get_yaml()
        )
```
we should use our own tools to do this, so the parse_yaml file could be deleted in the future and we hide all yaml parsing concise in the getter module... This avoids spreading external libs in the whole code base

Reply:
I changed it :+1: